CHANGES

```diff
@@ -3,6 +3,145 @@ repositories (changes that are automatically handled by the format upgrade tools
+  * `OpamClient.init` and `OpamClient.reinit`: now can have additional cygwin
+    packages to install [#5930 @moyodiallo]
+  * Expose `OpamSolution.print_depext_msg` [#5994 @dra27]
+  * Extracted `OpamSolution.install_sys_packages` from
+    `OpamSolution.install_depexts` [#5994 @dra27]
+  * `OpamInitDefaults.required_packages_for_cygwin`: no longer includes git;
+    as the need to add that is computed in `OpamClient` [#6000 @dra27]
+  * `OpamDownload.download_command`: separate output from stdout and stderr
+    [#5984 @kit-ty-kate]
+  * `OpamEnv.cygwin_non_shadowed_programs`: exposes the list of executables
+    (not including git) which should always come from Cygwin [#6000 @dra27]
+  * `opamSysInteract.Cygwin.install`: de-label `packages` argument
+    [#6000 @dra27]
+  * `OpamSysInteract.Cygwin.check_install` renamed to `analyse_install` which
+    now also returns whether the installation found was MSYS2 or Cygwin
+    [#6000 @dra27]
+  * `OpamPath`: remove `OpamPath.Switch.last_env` function in favor to
+    `OpamPath.last_env` as the files are no more stored in switch directory
+    [#5962 @moyodiallo - fix #5823]
+  * `OpamFilter.map_up`: correct handling of FDefined [#5983 @dra27]
+  * `OpamFilter.fold_down_left`: correct handling of FDefined and FUndef
+    [#5983 @dra27]
+  * `OpamStd.String`: add `split_quoted` that preserves quoted separator
+    [#5935 @dra27]
+  * `OpamSystem.copy_dir` and `OpamSystem.mv` may display a warning on Windows
+    if an invalid symlink (e.g. an LXSS Junction) is found [#5953 @dra27]
+  * `OpamStubs.getVersionInfo`: on Windows, retrives the version information
+    block of an executable/library [#5963 @dra27]
+  * `OpamStubs.readRegistry`: on Windows, complements `OpamStubs.writeRegistry`
+    [#5963 @dra27]
+  * `OpamStubs.get_initial_environment`: on Windows, returns the pristine
+    environment for new shells [#5963 @dra27]
+  * `OpamConsole`: Add `formatted_errmsg` [#5999 @kit-ty-kate]
+  * `OpamConsole.menu` now supports up to 35 menu items [#5992 @dra27]
+  * `OpamStd.Sys.resolve_command`: extracted the logic from
+    `OpamSystem.resolve_command`, without the default environment handling from
+    OpamProcess. [#5991 @dra27]
+  * `OpamStd.Sys.resolve_in_path`: split the logic of
+    `OpamStd.Sys.resolve_command` to allow searching for an arbitrary file in
+    the search path [#5991 @dra27]
+  * `OpamProcess.run_background`: name the stderr output file have the .err
+    extension when cmd_stdout is given [#5984 @kit-ty-kate]
+  * `OpamStubs.enumRegistry`: on Windows, retrieves all the values of a given
+    type from a registry key, with their names [#6000 @dra27]
+  * `OpamCompat`: add `Seq.find_map` from OCaml 4.14 [#6000 @dra27]
+  * `OpamStd.Sys.{get_windows_executable_variant,get_cygwin_variant,
+    is_cygwin_variant}`: renamed `~cygbin` to `?search_in_path` with a change
+    in semantics so that it acts as though the directory was simply the first
+    entry in PATH [#6000 @dra27]
```

Comment:
```suggestion
  * `OpamClient.init` and `OpamClient.reinit`: now can have additional cygwin
    packages to install [#5930 @moyodiallo]
  * `OpamInitDefaults.required_packages_for_cygwin`: no longer includes git; as
    the need to add that is computed in `OpamClient` [#6000 @dra27]
  * `OpamSolution`: expose `print_depext_msg` [#5994 @dra27]
  * `OpamSolution`: extract `install_sys_packages` from
    `OpamSolution.install_depexts` [#5994 @dra27]
  * `OpamDownload.download_command`: separate output from stdout and stderr
    [#5984 @kit-ty-kate]
  * `OpamEnv.cygwin_non_shadowed_programs`: exposes the list of executables
    (not including git) which should always come from Cygwin [#6000 @dra27]
  * `OpamSysInteract.Cygwin.check_install` renamed to `analyse_install` which
    now also returns whether the installation found was MSYS2 or Cygwin
    [#6000 @dra27]
  * `OpamSysInteract.Cygwin.install`: de-label `packages` argument
    [#6000 @dra27]
  * `OpamFilter.fold_down_left`: correct handling of `FDefined` and `FUndef`
    [#5983 @dra27]
  * `OpamFilter.map_up`: correct handling of FDefined [#5983 @dra27]
  * `OpamPath`: remove `OpamPath.Switch.last_env` function in favor to
    `OpamPath.last_env` as the files are no more stored in switch directory
    [#5962 @moyodiallo - fix #5823]
  * `OpamCompat`: add `Seq.find_map` from OCaml 4.14 [#6000 @dra27]
  * `OpamConsole.menu` now supports up to 35 menu items [#5992 @dra27]
  * `OpamConsole`: Add `formatted_errmsg` [#5999 @kit-ty-kate]
  * `OpamProcess.run_background`: name the stderr output file have the .err
    extension when cmd_stdout is given [#5984 @kit-ty-kate]
  * `OpamStd.String`: add `split_quoted` that preserves quoted separator
    [#5935 @dra27]
  * `OpamStd.Sys.resolve_command`: extracted the logic from
    `OpamSystem.resolve_command`, without the default environment handling from
    OpamProcess. [#5991 @dra27]
  * `OpamStd.Sys.resolve_in_path`: split the logic of
    `OpamStd.Sys.resolve_command` to allow searching for an arbitrary file in the
    search path [#5991 @dra27]
  * `OpamStd.Sys.{get_windows_executable_variant,get_cygwin_variant,
    is_cygwin_variant}`: renamed `~cygbin` to `?search_in_path` with a change in
    semantics so that it acts as though the directory was simply the first entry
    in PATH [#6000 @dra27]
  * `OpamStubs.enumRegistry`: on Windows, retrieves all the values of a given
    type from a registry key, with their names [#6000 @dra27]
  * `OpamStubs.getVersionInfo`: on Windows, retrives the version information
    block of an executable/library [#5963 @dra27]
  * `OpamStubs.get_initial_environment`: on Windows, returns the pristine
    environment for new shells [#5963 @dra27]
  * `OpamStubs.readRegistry`: on Windows, complements `OpamStubs.writeRegistry`
    [#5963 @dra27]
  * `OpamSystem.copy_dir` and `OpamSystem.mv` may display a warning on Windows
    if an invalid symlink (e.g. an LXSS Junction) is found [#5953 @dra27]
```